### DISCUSSION TITLE
Add pseudo selector wrap

### DISCUSSION THREAD

**Discussion by Arturexe** [tailwindlabs/tailwindcss#10365] [Ideas]
I know, this has been discussed in the past:

https://stackoverflow.com/questions/74166688/any-way-to-group-pseudo-selectors-in-tailwind
https://twitter.com/adamwathan/status/1461519820411789314

Basically what I want is to be able to instead of this:

`hover:bg-slate-200 hover:border-slate-200`

do this:

`hover:[bg-slate-200 border-slate-200]`

or some other form of it. 

Could you elaborate why this can't be done currently? Wouldn't it be possible to just use `hover:[]` as a namespace for all `hover:` pseudo selector classes?

It would make inline classes more readable if you could stack `pseudo selector classes`.

**Comment by zslabs** [comment]
What more elaboration would you need aside from what Adam mentions in the Twitter thread? This is the takeaway https://twitter.com/adamwathan/status/1461700281524277253?s=46&t=xWRDy9jPhMgiw7hpwUXa_w

**Reply by Arturexe** [reply]
Why would a namespace that unwraps everything into vanilla tailwind code cause a bigger css file and lower performance?

In addition, he said that he is hesitant, not that it will never be implemented.

Now's the time!

**Reply by wongjn** [reply]
> that unwraps everything into vanilla tailwind code

This would mean Tailwind would have to somehow "know" about the source code more than it does currently (as just a set of strings) and be able to modify the classes declared in said code out into the constituent parts. A significant point of Tailwind is that it is language-agnostic and this transformation could break projects, or perhaps there could be some projects where Tailwind simply cannot interact with the code beyond reading it as a plain string.

**Reply by Arturexe** [reply]
So a simple string transformation layer would be too difficult to implement? Does it not do that already to some degree?
